src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -624,12 +642,62 @@ private File handleResponse(Response response, String currentInstalledVersion, F
         } else if (response.getStatusCode() == HttpStatus.SC_NOT_MODIFIED) {
             logger.info("{} Present Detect installation is up to date - skipping download.", LOG_PREFIX);
         } else {
+            String problemUrl = captureProblemDetectUrl(downloadUrl);
             String message = StringUtils.isNotBlank(response.getStatusMessage()) ? response.getStatusMessage() : EnglishReasonPhraseCatalog.INSTANCE.getReason(response.getStatusCode(), Locale.ENGLISH);
-            logger.warn("{} Unable to download artifact. Response code: {} {}", LOG_PREFIX, response.getStatusCode(), message);
+            logger.warn("{} Unable to download artifact from {}. Response code: {} {}", LOG_PREFIX, problemUrl, response.getStatusCode(), message);
```

Comment:
Given that the response code here will be a 2xx level code for a valid URL that may not point to Detect, would it be better to not report any status code in the warning message to avoid confusion? I've seen status codes to be typically shown only in logger.traces in Detect so that's an option if we desire to have it somewhere for debugging.

Reply:
Are you saying have something like:
logger.warn("{} Unable to download artifact from {}.", ...
logger.trace("{} Response code: {} {}.", ...

in other words splitting into two statements?  Say something failed and from where (in the default logging) and have a trace to capture the status code? If so I think I like that...the warning with a 200 seems a bit odd to me as well.

Reply:
Yes, exactly like above was my thought too